core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterBuilderService.java

```diff
@@ -173,6 +173,11 @@ public void buildCluster(Long stackId) throws CloudbreakException, ClusterClient
                 proxyConfig.orElse(null),
                 template);
         clusterService.save(cluster);
+        // There are not post-install recipes by default. Post-install doesn't seem to check the recipe list,
+        //  and executes always, which consistently adds 10s.
+        // TODO Question: Is there a reason for this, or is it safe to switch this to
+        //  recipeEngine.executePostInstallRecipes(stack, hostGroupService.getRecipesByCluster(cluster.getId()));
```

Comment:
Good point. I don't think we need this phase if there are no recipes.

Reply:
Will add this to the patch.